src/ucs/sys/event_set.c

```diff
@@ -134,31 +135,43 @@ ucs_status_t ucs_event_set_del(ucs_sys_event_set_t *event_set, int event_fd)
     return UCS_OK;
 }
 
-ucs_status_t ucs_event_set_wait(ucs_sys_event_set_t *event_set, int timeout_ms,
+ucs_status_t ucs_event_set_wait(ucs_sys_event_set_t *event_set,
+                                unsigned max_events, int timeout_ms,
                                 ucs_event_set_handler_t event_set_handler,
-                                void *arg)
+                                void *arg, unsigned *read_events)
 {
-    int nready;
     struct epoll_event ep_events[UCS_EVENT_EPOLL_MAX_EVENTS];
-    int i;
+    unsigned i, max_wait_events;
+    int nready, events;
 
-    if (event_set_handler == NULL) {
-        return UCS_ERR_INVALID_PARAM;
-    }
+    ucs_assert(event_set_handler != NULL);
 
-    nready = epoll_wait(event_set->epfd, ep_events, UCS_EVENT_EPOLL_MAX_EVENTS,
+    max_wait_events = ucs_min(max_events, UCS_EVENT_EPOLL_MAX_EVENTS);
+    nready = epoll_wait(event_set->epfd, ep_events, max_wait_events,
                         timeout_ms);
-    if ((nready < 0) && (errno != EINTR)) {
-        ucs_error("epoll_wait() failed: %m");
+    if (nready < 0) {
+        *read_events = 0;
+
+        if (errno == EINTR) {
+            return UCS_INPROGRESS;
+        }
+        ucs_fatal("epoll_wait() failed: %m");
```

Comment:
error would be enough, fatal cause abort

Reply:
Changed to `ucs_error()`